modules/openapi-generator/src/main/resources/python/python-experimental/api_client.mustache

```diff
@@ -542,3 +566,159 @@ class ApiClient(object):
                     raise ApiValueError(
                         'Authentication token must be in `query` or `header`'
                     )
+
+{{#hasHttpSignatureMethods}}
+    def get_http_signature_headers(self, resource_path, method, headers, body, query_params):
+        """
+        Create a cryptographic message signature for the HTTP request and add the signed headers.
+        
+        :param resource_path : A string representation of the HTTP request resource path.
+        :param method: A string representation of the HTTP request method, e.g. GET, POST.
+        :param headers: A dict containing the HTTP request headers.
+        :param body: The string representation of the HTTP request body.
+        :param query_params: A string representing the HTTP request query parameters.
+        :return: A dict of HTTP headers that must be added to the outbound HTTP request.
+        """
+        if method is None:
+            raise Exception("HTTP method must be set")
+        if resource_path is None:
+            raise Exception("Resource path must be set")
+        if body is None:
```

Comment:
How about refactoring lines 586 to 628 into a function called get_signed_header_info?
All of that code is only focused on generating those two dictionaries
Like so:
`signed_headers, signed_headers_dict = get_signed_header_info(body, self.configuration, method)`
That would shorten our get_http_signature_headers and help make it more understandable.

Reply:
done.